sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/AsyncKeyResolver.java

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.keys.cryptography;
+
+import com.azure.core.credentials.TokenCredential;
+import com.azure.core.cryptography.AsyncKeyEncryptionKey;
+import com.azure.core.cryptography.AsyncKeyEncryptionKeyResolver;
+import com.azure.core.cryptography.KeyEncryptionKey;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.Configuration;
+import reactor.core.publisher.Mono;
+
+/**
+ * An object capable of asynchronously retrieving key encryption keys from a provided key identifier.
+ */
+public class AsyncKeyResolver implements AsyncKeyEncryptionKeyResolver {
```

Comment:
Discussed a bunch of changes to the two 'Resolver' classes offline. Please consider those as well.

Reply:
made the changes.